orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -88,8 +86,7 @@
       options.fileSystem(((HadoopOutputFile) file).getFileSystem());
     }
     options.setSchema(orcSchema);
-    this.writer = newOrcWriter(file, options, metadata);
-
+    this.writer = ORC.newFileWriter(file, options, metadata);
```

Comment:
is the `options.fileSystem` call still needed above?

Reply:
Yes that is still needed. We continue to retain the current behavior for HadoopIO. So the FileSystem from HadoopOutputFile or HadoopInputFile is used for the FS operations. When we use anything other than HadoopIO the FileIOFS kicks in.